### PR TITLE
fix: recommend `teleportAsync` usage over `teleport`

### LINKED ISSUE #395: explain when and why to use the async teleportation methods
first time seeing this page and idk if in scope, but we really should explain when and why to use the async methods (i.e. no sync chunk loads, performance 📈)

_Originally posted by @MiniDigger in https://github.com/PaperMC/docs/pull/394#discussion_r1615144265_

### PR DESCRIPTION
**Pull request by zlataovce**
Fixes #395.

### CODE REVIEW

File: docs/paper/dev/api/entity-api/entity-teleport.mdx [hunk 1]
```diff
@@ -52,7 +59,7 @@ All available teleport flags can be found in the <Javadoc name={"io.papermc.pape
 Teleport a player relatively, preventing velocity from being reset in the X, Y and Z axes.
 
 ```java
-player.teleport(
+player.teleportAsync(
```
Comment (Machine-Maker) [inline docs/paper/dev/api/entity-api/entity-teleport.mdx:62]: afaik, `teleportAsync` also has to be run on the main thread, it just won't block the main thread while loading chunks for the location.

**Review by Machine-Maker — CHANGES_REQUESTED** [review]

File: docs/paper/dev/api/entity-api/entity-teleport.mdx (hunk 1 above)
Comment (zlataovce) [reply docs/paper/dev/api/entity-api/entity-teleport.mdx:62]: I looked into the patch and didn't see a catcher, will add a warning box though

Comment (Machine-Maker) [reply docs/paper/dev/api/entity-api/entity-teleport.mdx:62]: I could be wrong about this, cause now it looks like the getChunkAtAsync method (which is what teleportAsync uses) does have separate logic if called off the main thread.

Comment (zlataovce) [reply docs/paper/dev/api/entity-api/entity-teleport.mdx:62]: tried running `teleportAsync` in `ForkJoinPool#commonPool` and joined the future, worked just fine

`teleport` failed as expected:
```
[19:10:15 WARN]: Exception in thread "ForkJoinPool.commonPool-worker-1" java.lang.IllegalStateException: PlayerTeleportEvent may only be triggered synchronously.
[19:10:15 WARN]: 	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:42)
[19:10:15 WARN]: 	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:131)
[19:10:15 WARN]: 	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:627)
[19:10:15 WARN]: 	at org.bukkit.craftbukkit.entity.CraftPlayer.teleport(CraftPlayer.java:1450)
[19:10:15 WARN]: 	at org.bukkit.craftbukkit.entity.CraftPlayer.teleport(CraftPlayer.java:1348)
[19:10:15 WARN]: 	at org.bukkit.craftbukkit.entity.CraftEntity.teleport(CraftEntity.java:226)
[19:10:15 WARN]: 	at test-plugin-1.0.0-SNAPSHOT.jar//me.kcra.testplugin.TestPlugin.lambda$onEnable$9(TestPlugin.java:183)
[19:10:15 WARN]: 	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1423)
[19:10:15 WARN]: 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
[19:10:15 WARN]: 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
[19:10:15 WARN]: 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
[19:10:15 WARN]: 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
[19:10:15 WARN]: 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
```

File: docs/paper/dev/api/entity-api/entity-teleport.mdx [hunk 2]
```diff
@@ -9,13 +9,27 @@ Entities can be instantaneously teleported to specific positions, synchronously
...
+:::danger[Performance]
+
+It is strongly recommended to use the `teleportAsync` API in most cases, as it avoids doing
+synchronous chunk loads, which put a lot of stress on the server's main thread - hurting
+overall performance.
+
+However, if you know that the chunk you're teleporting into is loaded, _it is recommended
+to use `teleport`_ to prevent the parallelization overhead.
+
+:::
```
Comment (olijeffers0n) [inline docs/paper/dev/api/entity-api/entity-teleport.mdx:21]:
```suggestion
:::note[Performance]

We recommend using `teleportAsync` when you expect to be teleporting into unloaded chunks. Teleporting 
synchronously *will* have an impact on server performance when called from the main thread, however using
`teleportAsync` does come with the caveat of all subsequent code being called from within the `onAccept`, 1 tick later.

For simple teleports within nearby (loaded) chunks, the performance different is small enough to justify using either.

:::
```

**Review by olijeffers0n — APPROVED** [review]

**Comment by MiniDigger** [comment]
thanks

**Pull request merged**